scipio/src/file_stream.rs

```diff
@@ -793,6 +824,7 @@ impl StreamWriterState {
                 }
 
                 let mut state = state.borrow_mut();
+                state.adjust_flushed_pos(file_pos);
```

Comment:
Is there a way to make this lazy? I'm concerned about the cost of this for writers that will never care about this information.

Reply:
Usually I am not too concerned about bookkeeping for writes: writes come in 128kB requests that take a lot of time to come back.

How would you do it lazily, though? I am open if you have a good suggestion.
The `file_pos` advances as we initiate the flush, so that is not good enough. We can keep an array of returned `file_pos` but if we don't trim it here it will keep growing which is worse: you win in time for the insert but lose in memory utilization.